checker.h

```diff
@@ -0,0 +1,17 @@
+#include <stdio.h>
+#include <assert.h>
+
+#define TempMAX 45
+#define TempMIN 0
+#define SOCMAX 80
+#define SOCMIN 20
+#define ChangeRate 0.8
+#define TRUE 1
```

Comment:
It's a good practice to give meaning to 1 and 0, as you have done here.
You could improve further, giving specific names instead of `TRUE` and `FALSE`. In this case, `IN_RANGE` and `OUT_OF_RANGE`?

Reply:
Yes i have updated the code with the comments